portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -72,11 +72,32 @@ private String getParameter(HttpServletRequest request, String name, String defa
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form.
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+     
+      if( userService.getCurrentUser() == null){
+          response.sendRedirect("/signIn");
+      }
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      
+    }else {
+      response.sendRedirect("/signIn");
+    }
+    response.sendRedirect("/signIn");
     String name = request.getParameter("name");
     String comment = request.getParameter("comment-box");
     String title = request.getParameter("comment-title");
-    String json = convertToJson(name,title,comment);
-    enteries.add(json);
-    response.sendRedirect("/index.html");
+
+    //create datastore and insert comments
+    Entity commentEntry = new Entity("Comment");
+    commentEntry.setProperty("name", name);
+    commentEntry.setProperty("comment", comment);
+    commentEntry.setProperty("title", title);
+    commentEntry.setProperty("email", userService.getCurrentUser().getEmail());
```

Comment:
I'm not sure if you confirmed this yourself, but this is where I get a NullPointerException when I click the Submit button without logging in first, and from the code flow, it looks like that's expected, because the `response.sendRedirect("/signIn")` statement doesn't finish the execution of this function. It keeps going until the closing curly brace or until it sees a `return` statement. If you wanted to stop right after redirect, put `return` after the `response.sendRedirect("/signIn")` statement.